cpp/include/raft/core/mdarray.hpp

```diff
@@ -74,7 +74,7 @@ using detail::vector_extent;
 template <typename IndexType>
 using extent_1d = vector_extent<IndexType>;
 
-template <typename IndexType>
+template <typename IndexType = std::uint32_t>
```

Comment:
Would you consider getting this default index type from some common source, to avoid code duplication and possible errors between `matrix_extent` and (say) `vector_extent`?

Reply:
There were several issues after merging with the introduced IndexType, I did not mean to change that here.